icepyx/core/read.py

```diff
@@ -344,54 +343,37 @@ class Read(EarthdataAuthMixin):
 
     def __init__(
         self,
-        data_source=None,  # DevNote: Make this a required arg when catalog is removed
+        data_source,
+        glob_kwargs={},
+        out_obj_type=None,  # xr.Dataset,
+        # deprecated arguments
         product=None,
         filename_pattern=None,
         catalog=None,
-        glob_kwargs={},
-        out_obj_type=None,  # xr.Dataset,
     ):
-        # Raise error for deprecated argument
-        if catalog:
-            raise DeprecationError(
-                "The `catalog` argument has been deprecated and intake is no longer supported. "
-                "Please use the `data_source` argument to specify your dataset instead."
-            )
-
         if data_source is None:
```

Comment:
```suggestion
```
Now that `data_source=None` Python will enforce this check

Reply:
Good catch!